pkg/pf/internal/plugin/provider_server.go

```diff
@@ -561,10 +561,11 @@ func (p *providerServer) Construct(ctx context.Context,
 		}
 		propertyDependencies[resource.PropertyKey(name)] = urns
 	}
+	protect := req.GetProtect()
 	options := pl.ConstructOptions{
 		Aliases:              aliases,
 		Dependencies:         dependencies,
-		Protect:              req.GetProtect(),
+		Protect:              &protect,
```

Comment:
GetProtect() papers over the nil I think.  A bit dangerous here.

Reply:
GetProtect() does have a nil check as of [this change](https://github.com/pulumi/pulumi/pull/18924/files#diff-e38a8413bb3b6d73725cb4349bdb60867cc13b78ac22b4efa722812972f7e7b6R2669)

Regardless, it looks like we can call the field directly now, so we can do that instead.